R/init.R

```diff
@@ -307,3 +310,34 @@ renv_init_repos <- function() {
   repos
 
 }
+
+renv_init_type <- function(project) {
+
+  # if we don't have a DESCRIPTION file, use the default
+  if (!file.exists(file.path(project, "DESCRIPTION")))
+    return(settings$snapshot.type(project = project))
+
+  # otherwise, ask the user if they want to explicitly enumerate their
+  # R package dependencies in the DESCRIPTION file
+  choice <- menu(
+
+    title = c(
+      "This project contains a DESCRIPTION file.",
+      "DESCRIPTION files are often used to explicitly declare a project's R dependencies.",
+      "How would you like to declare R package dependencies in this project?"
+    ),
+
+    choices = c(
+      explicit = "I will record my project's package dependencies in the DESCRIPTION file.     (explicit mode)",
+      implicit = "I will let renv automatically determine the project's package dependencies.  (implicit mode)"
```

Comment:
```suggestion
      explicit = "Take project dependencies only from the DESCRIPTION (explicit mode)",
      implicit = "Use all files to determine the dependencies (implicit mode)"
```

That's probably a bit too short, but in particularly the "I will" reads weirdly to me.

Reply:
Thanks! This seems like an improvement to me.

The main thing I was trying to decide was whether we need to be clear that it's the user's responsibility to update the DESCRIPTION file, but maybe that's going to be obvious enough for anyone who has taken the time to actually create one?

Reply:
Yeah, I think it's obvious enough from the context + message.